File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/ComplexModel.java
package com.raizlabs.android.dbflow.test.structure;

import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ForeignKey;
import com.raizlabs.android.dbflow.annotation.ForeignKeyReference;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;
import com.raizlabs.android.dbflow.test.TestDatabase;

@Table(database = TestDatabase.class)
public class ComplexModel extends BaseModel {

    @Column
    @PrimaryKey
    String name;

    @Column
    @ForeignKey(saveForeignKeyModel = true, references = {@ForeignKeyReference(columnName = "testmodel_id",
            columnType = String.class, foreignKeyColumnName = "name")})
    TestModel1 testModel1;

    @Column
    @ForeignKey(saveForeignKeyModel = true, references = {@ForeignKeyReference(columnName = "mapmodel_id",
            columnType = String.class, foreignKeyColumnName = "name")})
    TestModel2 mapModelContainer;

}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/test/structure/ComplexModel.kt
package com.raizlabs.android.dbflow.test.structure

import com.raizlabs.android.dbflow.annotation.*
import com.raizlabs.android.dbflow.structure.BaseModel
import com.raizlabs.android.dbflow.test.TestDatabase

@Table(database = TestDatabase::class)
class ComplexModel : BaseModel() {

    @Column
    @PrimaryKey
    var name: String? = null

    @Column
    @ForeignKey(saveForeignKeyModel = true,
        references = arrayOf(ForeignKeyReference(columnName = "testmodel_id",
            foreignKeyColumnName = "name")))
    var testModel1: TestModel1? = null

    @Column
    @ForeignKey(saveForeignKeyModel = true,
        references = arrayOf(ForeignKeyReference(columnName = "mapmodel_id",
            foreignKeyColumnName = "name")))
    var mapModelContainer: TestModel2? = null

}
